src/runtime_src/core/pcie/driver/linux/xocl/userpf/xocl_kds.c

```diff
@@ -315,6 +331,26 @@ static int xocl_command_ioctl(struct xocl_dev *xdev, void *data,
 	xcmd->cb.notify_host = notify_execbuf;
 	xcmd->gem_obj = obj;
 
+	if (in_kernel) {
+		struct drm_xocl_execbuf_cb *args_cb =
+					(struct drm_xocl_execbuf_cb *)data;
```

Comment:
This is not a must resolve thing.
To make this work, the drm_xocl_execbuf_cb must be an extend of drm_xocl_execbuf. For the short term and for internal quick fix, this is okay to me.
But, ideally, we should not use two structs.
Maybe we can accept this code but we definitely need a little bit refine in the future.

Reply:
I had thought of it. But the problem I see is the callback function and data pointer may look out of place in user-space struct (drm_xocl_execbuf).  Can we think over the options and make a subsequent check-in?